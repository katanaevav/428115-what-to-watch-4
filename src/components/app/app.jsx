import React, {PureComponent} from "react";
import {Switch, Route, BrowserRouter} from "react-router-dom";
import PropTypes from "prop-types";
import Main from "../main/main.jsx";
import MoviePage from "../movie-page/movie-page.jsx";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer.js";
import {getFilteredMovies} from "../../selectors.js";
import {Screens, MAX_SIMILAR_MOVIES_COUNT} from "../../const.js";
import withMovieTabs from "../../hoc/with-movie-tabs/with-movie-tabs.js";
import CinemaScreen from "../cinema-screen/cinema-screen.jsx";

const MoviePageWrapper = withMovieTabs(MoviePage);

class App extends PureComponent {
  constructor(props) {
    super(props);

    this._movieCardMouseOverHandler = this._movieCardMouseOverHandler.bind(this);
  }

  _getMovieById(movieId) {
    const {movies} = this.props;

    return movies.find((movie) => movie.id === Number.parseInt(movieId, 10));
  }

  _movieCardMouseOverHandler() {}


  _renderApp() {
    const {promoMovie, movies, genres, currentGenreFilter, onMovieFilterClick, onMovieTitleClick, selectedMovieId, currentPage} = this.props;
    const {title: promoMovieTitle, genre: promoMovieGenre, year: promoMovieYear, bigPoster: promoMoviePoster, cover: promoMovieCover} = promoMovie;

    switch (currentPage) {
      case Screens.MOVIE_PAGE_SCREEN:
        const {id, title, genre, year, runTime, bigPoster, cover, ratingScore, ratingCount, directors, starrings, descriptions, reviews} = this._getMovieById(selectedMovieId);
        const similarMovies = movies.filter((movie) => (movie.genre === genre) && (movie.id !== id)).slice(0, MAX_SIMILAR_MOVIES_COUNT);
        return (
          <MoviePageWrapper
            id = {id}
            title = {title}
            genre = {genre}
            year = {year}
            runTime = {runTime}
            bigPoster = {bigPoster}
            cover = {cover}
            ratingScore = {ratingScore}
            ratingCount = {ratingCount}
            directors = {directors}
            starrings = {starrings}
            descriptions = {descriptions}
            reviews = {reviews}
            similarMovies = {similarMovies}
            onMovieTitleClick = {onMovieTitleClick}
          />
        );

      default:
        return (
          <Main
            promoMovieTitle = {promoMovieTitle}
            promoMovieGenre = {promoMovieGenre}
            promoMovieYear = {promoMovieYear}
            promoMovieCover = {promoMovieCover}
            promoMovieBigPoster = {promoMoviePoster}
            genres = {genres}
            movies = {movies}
            currentGenreFilter = {currentGenreFilter}
            onMovieTitleClick = {onMovieTitleClick}
            onMovieFilterClick = {onMovieFilterClick}
          />
        );
    }
  }

  render() {
    const {promoMovie} = this.props;
    return (
      <BrowserRouter>
        <Switch>
          <Route exact path="/">
            {this._renderApp()}
          </Route>
          <Route exact path="/dev-film">
            <MoviePageWrapper
              id = {0}
              title = {promoMovie.title}
              genre = {promoMovie.genre}
              year = {promoMovie.year}
              runTime = {promoMovie.runTime}
              bigPoster = {promoMovie.bigPoster}
              cover = {promoMovie.cover}
              ratingScore = {promoMovie.ratingScore}
              ratingCount = {promoMovie.ratingCount}
              directors = {promoMovie.directors}
              starrings = {promoMovie.starrings}
              descriptions = {promoMovie.descriptions}
              reviews = {promoMovie.reviews}
              similarMovies = {this.props.movies}
              onMovieTitleClick = {() => {}}
            />
          </Route>
          <Route exact path="/dev-player">
            <CinemaScreen />
          </Route>
        </Switch>
      </BrowserRouter>
    );
  }

}

App.propTypes = {
  promoMovie: PropTypes.shape().isRequired,
  movies: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        smallPoster: PropTypes.string.isRequired,
        genre: PropTypes.string.isRequired,
        year: PropTypes.number.isRequired,
      })).isRequired,
  genres: PropTypes.array.isRequired,
  currentGenreFilter: PropTypes.string.isRequired,
  onMovieFilterClick: PropTypes.func.isRequired,
  onMovieTitleClick: PropTypes.func.isRequired,
  selectedMovieId: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
};

const mapStateToProps = (state) => ({
  currentGenreFilter: state.currentGenreFilter,
  promoMovie: state.PromoMovie,
  movies: getFilteredMovies(state.movies, state.currentGenreFilter),
  genres: state.genres,
  selectedMovieId: state.selectedMovieId,
  currentPage: state.currentPage,
});

const mapDispatchToProps = (dispatch) => ({
  onMovieFilterClick(filterName) {
    dispatch(ActionCreator.setCurrentFilter(filterName));
  },

  onMovieTitleClick(movieId) {
    dispatch(ActionCreator.openMovieScreen(movieId));
  }
});


export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
