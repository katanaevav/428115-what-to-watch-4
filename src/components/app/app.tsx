import * as React from "react";
import {Switch, Route, Router, Redirect} from "react-router-dom";
import Main from "../main/main";
import MoviePage from "../movie-page/movie-page";
import SignIn from "../sign-in/sign-in";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer/state/state";
import {MAX_SIMILAR_MOVIES_COUNT, AppRoute, AuthorizationStatus} from "../../const";
import withMovieTabs from "../../hoc/with-movie-tabs/with-movie-tabs";
import withNewReview from "../../hoc/with-new-review/with-new-review";
import CinemaScreen from "../cinema-screen/cinema-screen";
import withCinemaVideoPlayer from "../../hoc/with-cinema-video-player/with-cinema-video-player";
import {getCurrentGenreFilter, getFilteredMovies} from "../../reducer/state/selectors";
import {getMovies, getPromoMovie, getGenres, getMovieComments, getSavingMovieCommentStatus, getSavingMovieFavoriteStatus, getMyMovies} from "../../reducer/data/selectors";
import {getAuthorizationStatus, getAvatarUrl} from "../../reducer/user/selectors";
import {Operation as UserOperation} from "../../reducer/user/user";
import {Operation as DataOperation} from "../../reducer/data/data";
import AddReview from "../add-review/add-review";
import history from "../../history";
import MyList from "../my-list/my-list";
import PrivateRoute from "../private-route/private-route";
import {Movie, Comment} from "../../types";


interface Props {
  authorizationStatus: string;
  login: () => void;
  avatarUrl?: string;
  savingMovieCommentStatus?: string;
  savingMovieFavoriteStatus?: string;
  promoMovie: Movie;
  movies: Array<Movie>;
  myMovies?: Array<Movie>;
  genres: Array<string>;
  currentGenreFilter: string;
  onMovieFilterClick: () => void;
  movieComments: Array<Comment>;
  getComments: () => void;
  saveComment: () => void;
  setFavoriteStatus: () => void;
}


const CinemaScreenWrapped = withCinemaVideoPlayer(CinemaScreen);
const AddReviewWrapped = withNewReview(AddReview);
const MoviePageWrapper = withMovieTabs(MoviePage);


class App extends React.PureComponent<Props, {}> {
  constructor(props) {
    super(props);

    this._getMovieById = this._getMovieById.bind(this);
  }

  _getMovieById(movieId) {
    const {movies} = this.props;

    return movies.find((movie) => movie.id === Number.parseInt(movieId, 10));
  }

  render() {
    const {
      movies,
      login,
      authorizationStatus,
      avatarUrl,
      myMovies,
      promoMovie,
      genres,
      currentGenreFilter,
      onMovieFilterClick,
      savingMovieFavoriteStatus,
      setFavoriteStatus
    } = this.props;

    if (movies.length && myMovies.length >= 0 && promoMovie) {
      return (
        <Router history = {history}>
          <Switch>
            <Route exact path = {AppRoute.ROOT}>
              <Main
                authorizationStatus = {authorizationStatus}
                avatarUrl = {avatarUrl}
                promoMovie = {promoMovie}
                genres = {genres}
                movies = {movies}
                currentGenreFilter = {currentGenreFilter}
                onMovieFilterClick = {onMovieFilterClick}
                savingMovieFavoriteStatus = {savingMovieFavoriteStatus}
                setFavoriteStatus = {setFavoriteStatus}
              />
            </Route>

            <Route exact path = {AppRoute.LOGIN}>
              {authorizationStatus === AuthorizationStatus.NO_AUTH ?
                <SignIn
                  onSubmit = {login}
                /> :
                <Redirect to = {AppRoute.ROOT} />}
            </Route>

            <PrivateRoute
              exact
              path = {AppRoute.MY_LIST}
              render = {() => {
                return (
                  <MyList
                    myMovies = {myMovies}
                    authorizationStatus = {authorizationStatus}
                    avatarUrl = {avatarUrl}
                  />
                );
              }}
            />

            <PrivateRoute
              path = {AppRoute.FILMS + AppRoute.ID + AppRoute.ADD_REVIEW}
              exact
              render = {(props) => {
                const selectedMovie = this._getMovieById(props.computedMatch.params.id);
                const {saveComment, savingMovieCommentStatus} = this.props;

                return (
                  <AddReviewWrapped
                    movie = {selectedMovie}
                    authorizationStatus = {authorizationStatus}
                    avatarUrl = {avatarUrl}
                    onSaveComment = {saveComment}
                    savingMovieCommentStatus = {savingMovieCommentStatus}
                    {...props}
                  />
                );
              }}
            />

            <Route
              path = {AppRoute.FILMS + AppRoute.ID}
              render = {(props) => {
                const {movieComments} = this.props;
                const selectedMovie = this._getMovieById(props.match.params.id);
                const {id, genre} = selectedMovie;
                const similarMovies = movies.filter((movie) => (movie.genre === genre) && (movie.id !== id)).slice(0, MAX_SIMILAR_MOVIES_COUNT);

                return (
                  <MoviePageWrapper
                    authorizationStatus = {authorizationStatus}
                    avatarUrl = {avatarUrl}
                    movie = {selectedMovie}
                    comments = {movieComments}
                    similarMovies = {similarMovies}
                    savingMovieFavoriteStatus = {savingMovieFavoriteStatus}
                    setFavoriteStatus = {setFavoriteStatus}
                    getComments = {this.props.getComments}
                  />
                );
              }}
            />

            <Route
              path = {AppRoute.PLAYER + AppRoute.ID}
              render = {
                (props) => {
                  return (
                    <CinemaScreenWrapped
                      movie = {this._getMovieById(props.match.params.id)}
                      onExitVideoPlayer = {() => {
                        history.goBack();
                      }}
                      {...props}
                    />
                  );
                }
              }
            />

          </Switch>
        </Router>
      );
    }

    return null;
  }
}


const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  currentGenreFilter: getCurrentGenreFilter(state),
  promoMovie: getPromoMovie(state),
  movies: getFilteredMovies(getMovies(state), getCurrentGenreFilter(state)),
  myMovies: getMyMovies(state),
  genres: getGenres(state),
  movieComments: getMovieComments(state),
  avatarUrl: getAvatarUrl(state),
  savingMovieCommentStatus: getSavingMovieCommentStatus(state),
  savingMovieFavoriteStatus: getSavingMovieFavoriteStatus(state),
});

const mapDispatchToProps = (dispatch) => ({
  setFavoriteStatus(favoriteStatus, action) {
    dispatch(DataOperation.saveMovieFavoriteStatus(favoriteStatus, action));
  },

  saveComment(comment, action) {
    dispatch(DataOperation.saveMovieComment(comment, action));
  },

  getComments(movieId) {
    dispatch(DataOperation.loadMovieComments(movieId));
  },

  login(authData, action) {
    dispatch(UserOperation.login(authData, action));
  },

  onMovieFilterClick(filterName) {
    dispatch(ActionCreator.setCurrentFilter(filterName));
  },
});


export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
