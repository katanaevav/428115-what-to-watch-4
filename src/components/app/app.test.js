import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Screens} from "../../const.js";
import {AuthorizationStatus} from "../../const.js";
import {SavingStatus} from "../../const.js";

const mockStore = configureStore([]);

const PromoMovie = {
  title: `The Grand Budapest Hotel`,
  smallPoster: `img/the-grand-budapest-hotel-poster.jpg`,
  genre: `Drama`,
  year: 2014,
  runTime: 170,
  bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
  cover: `img/bg-the-grand-budapest-hotel.jpg`,
  ratingScore: 8.9,
  ratingCount: 240,
  backgroundColor: `#AABBCC`,
  directors: [`Уэс Андерсон`],
  starrings: [`Рэйф Файнс`, `Тони Револори`, `Сирша Ронан`, `Эдриан Броуди`, `Уиллем Дефо`, `Эдвард Нортон`, `Матьё Амальрик`, `Харви Кейтель`, `Ф. Мюррэй Абрахам`, `Тильда Суинтон`],
  descriptions: [
    `Фильм рассказывает об увлекательных приключениях легендарного консьержа Густава и его юного друга, портье Зеро Мустафы.`,
    `Сотрудники гостиницы становятся свидетелями кражи и поисков бесценных картин эпохи Возрождения, борьбы за огромное состояние богатой семьи и… драматических изменений в Европе между двумя кровопролитными войнами XX века.`
  ],
  reviews: [
    {
      id: 0,
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      author: `Kate Muir`,
      date: Date.parse(`December 24, 2016`),
      mark: `4`,
    },
    {
      id: 1,
      text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
      author: `Paula Fleri-Soler`,
      date: Date.parse(`December 15, 2018`),
      mark: `5`,
    },
    {
      id: 2,
      text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
      author: `Amanda Greever`,
      date: Date.parse(`November 18, 2015`),
      mark: `3`,
    },
  ],
};

const NO_FILTER = `All genres`;

const GENRES = [`Drama`, `Fantasy`];

const Movies = [
  {
    id: 0,
    title: `Bohemian rhapsody`,
    smallPoster: `img/bohemian-rhapsody.jpg`,
    genre: `Drama`,
    year: 2018,
    runTime: 190,
    bigPoster: `img/bohemian-rhapsody.jpg`,
    cover: `img/bohemian-rhapsody.jpg`,
    ratingScore: 7.9,
    ratingCount: 284651,
    backgroundColor: `#AABBCC`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    directors: [`Брайан Сингер`],
    starrings: [`Рами Малек`, `Люси Бойнтон`, `Гвилим Ли`, `Бен Харди`, `Джозеф Маццелло`, `Эйдан Гиллен`, `Аллен Лич`, `Том Холландер`, `Майк Майерс`, `Аарон МакКаскер`],
    descriptions: [
      `Чествование группы Queen, их музыки и их выдающегося вокалиста Фредди Меркьюри, который бросил вызов стереотипам и победил условности, чтобы стать одним из самых любимых артистов на планете.`,
      `Фильм прослеживает головокружительный путь группы к успеху благодаря их культовым песням и революционному звуку, практически распад коллектива, поскольку образ жизни Меркьюри выходит из-под контроля, и их триумфальное воссоединение накануне концерта Live Aid, ставшим одним из величайших выступлений в истории рок-музыки.`,
    ],
  },
  {
    id: 1,
    title: `Moonrise kingdom`,
    smallPoster: `img/moonrise-kingdom.jpg`,
    genre: `Fantasy`,
    year: 2012,
    runTime: 185,
    bigPoster: `img/moonrise-kingdom.jpg`,
    cover: `img/moonrise-kingdom.jpg`,
    ratingScore: 8.9,
    ratingCount: 240,
    backgroundColor: `#AABBCC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    directors: [`Уэс Андерсон`],
    starrings: [`Брюс Уиллис`, `Эдвард Нортон`, `Билл Мюррей`, `Фрэнсис МакДорманд`, `Джаред Гилман`, `Кара Хэйуорд`, `Боб Бэлабан`, `Тильда Суинтон`, `Джейсон Шварцман`, `Харви Кейтель`],
    descriptions: [
      `60-е годы XX века. Пара влюблённых подростков, живущих на острове в Новой Англии, убегает из-под присмотра взрослых.`,
      `Сэм Шакаски — бойскаут, сирота, от которого отказались приемные родители, из-за своего непростого характера ставший изгоем среди других бойскаутов, и Сьюзи Бишоп — замкнутая двенадцатилетняя неуравновешенная девочка, живущая мечтами о волшебных мирах. После обнаружения пропажи местный шериф начинает расследование, а вожатый лагеря бойскаутов организует поисковый отряд.`,
    ],
  },
];

it(`Render App with Main screen`, () => {

  const store = mockStore({});

  const tree = renderer.create(
      <Provider store={store}>
        <App
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          login={() => {}}
          avatarUrl = {``}
          savingMovieCommentStatus = {SavingStatus.SUCCESS}
          promoMovie = {PromoMovie}
          movies = {Movies}
          genres = {GENRES}
          currentGenreFilter = {NO_FILTER}
          onMovieFilterClick = {() => {}}
          onMovieTitleClick = {() => {}}
          onPlayMovieClick = {() => {}}
          selectedMovieId = {-1}
          onAddReviewClick = {() => {}}
          currentPage = {Screens.MAIN_SCREEN}
          onExitVideoPlayer = {() => {}}
          movieComments = {[]}
          getComments = {() => {}}
          saveComment = {() => {}}
          onOpenAuthScreen = {() => {}}
          authMessage = {``}
        />
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render App with Movie page screen`, () => {

  const store = mockStore({});

  const tree = renderer.create(
      <Provider store={store}>
        <App
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          login={() => {}}
          avatarUrl = {``}
          savingMovieCommentStatus = {SavingStatus.SUCCESS}
          promoMovie = {PromoMovie}
          movies = {Movies}
          genres = {GENRES}
          currentGenreFilter = {NO_FILTER}
          onMovieFilterClick = {() => {}}
          onMovieTitleClick = {() => {}}
          onPlayMovieClick = {() => {}}
          selectedMovieId = {0}
          onAddReviewClick = {() => {}}
          currentPage = {Screens.MOVIE_PAGE_SCREEN}
          onExitVideoPlayer = {() => {}}
          movieComments = {[]}
          saveComment = {() => {}}
          getComments = {() => {}}
          onOpenAuthScreen = {() => {}}
          authMessage = {``}
        />
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
