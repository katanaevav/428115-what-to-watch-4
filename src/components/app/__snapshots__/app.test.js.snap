// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App with Main screen 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link "
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="/sprite.svg#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="/sprite.svg#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            data-key="Drama"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            data-key="Fantasy"
            href="#"
            onClick={[Function]}
          >
            Fantasy
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          data-key={0}
        >
          <div
            className="small-movie-card__image"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <video
              height="175"
              poster="img/bohemian-rhapsody.jpg"
              preload="metadata"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Bohemian rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-key={1}
        >
          <div
            className="small-movie-card__image"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <video
              height="175"
              poster="img/moonrise-kingdom.jpg"
              preload="metadata"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Moonrise kingdom
            </a>
          </h3>
        </article>
      </div>
      
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Render App with Movie page screen 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    data-key={0}
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Bohemian rhapsody"
          src="img/bohemian-rhapsody.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link "
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Bohemian rhapsody
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2018
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="/sprite.svg#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="/sprite.svg#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="add-review.html"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Bohemian rhapsody"
            height="327"
            src="img/bohemian-rhapsody.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  data-key={0}
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  data-key={1}
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  data-key={2}
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              7,9
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Good
              </span>
              <span
                className="movie-rating__count"
              >
                284651
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              Ð§ÐµÑÑÐ²Ð¾Ð²Ð°Ð½Ð¸Ðµ Ð³ÑÑÐ¿Ð¿Ñ Queen, Ð¸Ñ Ð¼ÑÐ·ÑÐºÐ¸ Ð¸ Ð¸Ñ Ð²ÑÐ´Ð°ÑÑÐµÐ³Ð¾ÑÑ Ð²Ð¾ÐºÐ°Ð»Ð¸ÑÑÐ° Ð¤ÑÐµÐ´Ð´Ð¸ ÐÐµÑÐºÑÑÑÐ¸, ÐºÐ¾ÑÐ¾ÑÑÐ¹ Ð±ÑÐ¾ÑÐ¸Ð» Ð²ÑÐ·Ð¾Ð² ÑÑÐµÑÐµÐ¾ÑÐ¸Ð¿Ð°Ð¼ Ð¸ Ð¿Ð¾Ð±ÐµÐ´Ð¸Ð» ÑÑÐ»Ð¾Ð²Ð½Ð¾ÑÑÐ¸, ÑÑÐ¾Ð±Ñ ÑÑÐ°ÑÑ Ð¾Ð´Ð½Ð¸Ð¼ Ð¸Ð· ÑÐ°Ð¼ÑÑ Ð»ÑÐ±Ð¸Ð¼ÑÑ Ð°ÑÑÐ¸ÑÑÐ¾Ð² Ð½Ð° Ð¿Ð»Ð°Ð½ÐµÑÐµ.
            </p>
            <p>
              Ð¤Ð¸Ð»ÑÐ¼ Ð¿ÑÐ¾ÑÐ»ÐµÐ¶Ð¸Ð²Ð°ÐµÑ Ð³Ð¾Ð»Ð¾Ð²Ð¾ÐºÑÑÐ¶Ð¸ÑÐµÐ»ÑÐ½ÑÐ¹ Ð¿ÑÑÑ Ð³ÑÑÐ¿Ð¿Ñ Ðº ÑÑÐ¿ÐµÑÑ Ð±Ð»Ð°Ð³Ð¾Ð´Ð°ÑÑ Ð¸Ñ ÐºÑÐ»ÑÑÐ¾Ð²ÑÐ¼ Ð¿ÐµÑÐ½ÑÐ¼ Ð¸ ÑÐµÐ²Ð¾Ð»ÑÑÐ¸Ð¾Ð½Ð½Ð¾Ð¼Ñ Ð·Ð²ÑÐºÑ, Ð¿ÑÐ°ÐºÑÐ¸ÑÐµÑÐºÐ¸ ÑÐ°ÑÐ¿Ð°Ð´ ÐºÐ¾Ð»Ð»ÐµÐºÑÐ¸Ð²Ð°, Ð¿Ð¾ÑÐºÐ¾Ð»ÑÐºÑ Ð¾Ð±ÑÐ°Ð· Ð¶Ð¸Ð·Ð½Ð¸ ÐÐµÑÐºÑÑÑÐ¸ Ð²ÑÑÐ¾Ð´Ð¸Ñ Ð¸Ð·-Ð¿Ð¾Ð´ ÐºÐ¾Ð½ÑÑÐ¾Ð»Ñ, Ð¸ Ð¸Ñ ÑÑÐ¸ÑÐ¼ÑÐ°Ð»ÑÐ½Ð¾Ðµ Ð²Ð¾ÑÑÐ¾ÐµÐ´Ð¸Ð½ÐµÐ½Ð¸Ðµ Ð½Ð°ÐºÐ°Ð½ÑÐ½Ðµ ÐºÐ¾Ð½ÑÐµÑÑÐ° Live Aid, ÑÑÐ°Ð²ÑÐ¸Ð¼ Ð¾Ð´Ð½Ð¸Ð¼ Ð¸Ð· Ð²ÐµÐ»Ð¸ÑÐ°Ð¹ÑÐ¸Ñ Ð²ÑÑÑÑÐ¿Ð»ÐµÐ½Ð¸Ð¹ Ð² Ð¸ÑÑÐ¾ÑÐ¸Ð¸ ÑÐ¾Ðº-Ð¼ÑÐ·ÑÐºÐ¸.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                ÐÑÐ°Ð¹Ð°Ð½ Ð¡Ð¸Ð½Ð³ÐµÑ
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Ð Ð°Ð¼Ð¸ ÐÐ°Ð»ÐµÐº, ÐÑÑÐ¸ ÐÐ¾Ð¹Ð½ÑÐ¾Ð½, ÐÐ²Ð¸Ð»Ð¸Ð¼ ÐÐ¸, ÐÐµÐ½ Ð¥Ð°ÑÐ´Ð¸
                 and others
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      />
      
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
